### DISCUSSION TITLE
Map int column to enum

### DISCUSSION THREAD

**Discussion by yibuw** [prisma/prisma#14907] [Q&A]
## Problem

I have an old DB, can use int with Enum, how to do it?

![image](https://user-images.githubusercontent.com/86776169/140026450-9e01f778-5ed9-4f33-a1b6-de40751e5805.png)

**Comment by janpio** [comment]
What exactly are you trying to do? 

`@map` is used to map Prisma Schema names to the actual name in the database. Even if the name was `0`, it would still be represented as a String on the database level, so `@map("0")` instead of `@map(0)`.

**Comment by BertBR** [comment]
There's any solution for this problem?

I'm facing the same issue while trying to map a enum value. Check this out:

`schema.prisma`
```ts
enum Invoice_type {
  subscription @map("0")
  rent         @map("1")
  sell         @map("2")
}
```

`prisma.client`
```ts
export const Invoice_type: {
  subscription: 'subscription',
  rent: 'rent',
  sell: 'sell'
};
```

**Reply by BigBrainAFK** [reply]
Is there a reason why Prisma doesn't use actual enums for this? As far as I recall TypeScript supports string Enums.
